CONTRIBUTING.md

```diff
@@ -0,0 +1,106 @@
+# Contributing
+
+Pull requests are welcome!
+
+## Getting started
+
+First [fork this repo](https://help.github.com/articles/fork-a-repo/), and
+then clone it & set the upstream repo:
+
+```bash
+git clone git@github.com:your-github-username/halyard.git
+
+   # in my case (username is lwander), the above command looks like:
+   # git clone git@github.com:lwander/halyard.git
+
+cd halyard
+
+git remote add-url upstream git@github.com:spinnaker/halyard.git
+
+  # the above command allows you to push patches to the upstream repo for 
+  # approval
+```
+
+## Configuring IntelliJ
+
+In the project's root directory run
+
+```bash
+./gradlew idea
+```
+
+and then point IntelliJ at the generated project. You shouldn't need to import
+this project, simply navigating to __File__ > __Open__ > _the project's root
+directory_ should be enough.
+
+## Running the daemon
+
+In the project's root directory run
+
+```bash
+./gradlew
+```
+
+and the daemon will build & run. The first time you do this it will take a few
+minutes as it fetches the dependencies.
+
+## Running the CLI
+
+In the project's root directory run
+
+```bash
+cd halyard-cli/
+
+make
+
+./hal --help
+```
+
+> WARNING! Since this project uses one gradle project for two executables, you
+> currently can't run `make` and `./gradlew` at the same time (one of the two
+> will likley hang indefinitely).
+
+## Project structure
+
+This is a brief overview of how the project is structured.
+
+### halyard-cli/
+
+All CLI related code is here. 
+
+ * `command.v1` defines the command structure and entrypoints into command 
+ processing.  For example, the behavior of `hal config` is defined in the
+ `ConfigCommand.java` class.
```

Comment:
I think a fully-qualified class name would be better here than `ConfigCommand.java`. `ConfigCommand.java` is a file not a class. So something like "... is defined in the `com.netflix.spinnaker.halyard.cli.command.v1.ConfigCommand` class.".